### PR TITLE
Make Healthchecker ping continuous

### PR DESCRIPTION
**Pull request by aswinsuryan**
When the Timeout value is set in the pinger, it is calcuated from the
time pinger starts and not the time interval between send and receive.
So now we are now using a threshold of packet count after which the connection
will be marked as an error, if the packet loss continues to be 100%.

Signed-off-by: Aswin Surayanarayanan <asuryana@redhat.com>

### CODE REVIEW

File: pkg/cableengine/healthchecker/pinger.go [hunk 1]
```diff
@@ -63,8 +63,16 @@ func (p *pingerInfo) sendPing() {
...
 	pinger.RecordRtts = false
-	// After 3 seconds stop waiting.
-	pinger.Timeout = timeout
+
+	pinger.OnSend = func(packet *ping.Packet) {
+		// Pinger will be stopped once the packet loss reaches threshold and
+		// if packet loss is not 100% recreates the Pinger without marking an error
+		if pinger.Statistics().PacketsSent-pinger.Statistics().PacketsRecv > threshold {
+			klog.Infof("Packet loss reached a threshold of %d and hence stopping the pinger for HealthCheckIP: %q",
+				threshold, p.healthCheckIP)
```
Comment (mangelajo) [inline pkg/cableengine/healthchecker/pinger.go:72]: But, why would we stop it?.

I believe we should keep it up until connection is back, or the cable connection is removed.

**Review by mangelajo — CHANGES_REQUESTED** [review]

File: pkg/cableengine/healthchecker/pinger.go (hunk 1 above)
Comment (aswinsuryan) [reply pkg/cableengine/healthchecker/pinger.go]: Done.

File: pkg/cableengine/healthchecker/pinger.go [hunk 2]
```diff
@@ -8,10 +8,9 @@ import (
 	"k8s.io/klog"
 )
 
-var waitTime time.Duration = 15 * time.Second
-var timeout = 3 * time.Second
+var waitTime = 15 * time.Second
 
-const allpackets = 100
+const threshold = 5
```
Comment (dfarrell07) [inline pkg/cableengine/healthchecker/pinger.go:13]: Maybe a better name would be something like `maxLostPackets` or `maxLosablePackets`?

Comment (aswinsuryan) [reply pkg/cableengine/healthchecker/pinger.go]: Done.

**Review by mangelajo — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]

**Pull request merged**